### PR TITLE
SONARPY-1326 UnusedLocalVariableCheck(1481) should raise with sequence unpacking

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**

### CODE REVIEW

File: python-checks/src/test/java/org/sonar/python/checks/UnusedLocalVariableCheckTest.java [hunk 1]
```diff
@@ -35,4 +36,19 @@ public void custom() {
     check.format = "(_|myignore)";
     PythonCheckVerifier.verify("src/test/resources/checks/unusedLocalVariableCustom.py", check);
   }
+
+  @Test
+  public void quickFixTest() {
```
Comment (sallaigy) [inline python-checks/src/test/java/org/sonar/python/checks/UnusedLocalVariableCheckTest.java:41]: _minor_ Test naming is all over the place in the code base, but we should try to stick to `snake_case`, and maybe indicate that the quick fix is only for sequence unpacking.
```suggestion
  public void tuple_quick_fix() {
```

Comment (sallaigy) [inline python-checks/src/test/java/org/sonar/python/checks/UnusedLocalVariableCheckTest.java:41]: Also, there is another edge case with the quick fix, where it slightly changes the semantics of the program:
```python
def using_tuples():
    x, y = (1, 2)
    y = 5
    print(x)
```
Replacing `y` with `_` is not erroneous, but it somewhat changes program semantics. I think we can document this case with another test.

File: python-checks/src/test/resources/checks/unusedLocalVariable.py [hunk 2]
```diff
@@ -47,7 +47,8 @@ def function_with_lambdas():
 #             ^
 
 def using_tuples():
-    x, y = (1, 2)
+    x, y = (1, 2) # Noncompliant {{Replace unused local variable "y" with "_".}}
```
Comment (sallaigy) [inline python-checks/src/test/resources/checks/unusedLocalVariable.py:50]: Note that the rule description also needs to be updated, see the "Exceptions" section https://sonarsource.github.io/rspec/#/rspec/S1481/python 

```
for _ in range(10):
    do_something()
username, login, password = auth # This now raises
do_something_else(username, login)
```

**Review by sallaigy — CHANGES_REQUESTED** [review]
Looks good to me! Could you also update the rule description before merging?

File: python-checks/src/test/java/org/sonar/python/checks/UnusedLocalVariableCheckTest.java (hunk 1 above)
Comment (maksim-grebeniuk-sonarsource) [reply python-checks/src/test/java/org/sonar/python/checks/UnusedLocalVariableCheckTest.java:41]: done

File: python-checks/src/test/resources/checks/unusedLocalVariable.py (hunk 2 above)
Comment (maksim-grebeniuk-sonarsource) [reply python-checks/src/test/resources/checks/unusedLocalVariable.py:50]: done

**Review by sallaigy — APPROVED** [review]
Looks good to me!
